src/main/java/net/snowflake/ingest/streaming/internal/ChannelCache.java

```diff
@@ -94,6 +96,10 @@ void invalidateChannelIfSequencersMatch(
     }
   }
 
+  List<SnowflakeStreamingIngestChannelInternal<?>> getChannels() {
+    return cache.values().stream().flatMap(v -> v.values().stream()).collect(Collectors.toList());
```

Comment:
We might have stall channels (channels that are not active for a certain time and will be auto cleaned up server side) or invalidated channels, so I'm not sure if it's a good idea to do this for all channels in cache, I think what we can do is to pass a list of channel objects into getAllLatestCommittedOffsetTokens?

Reply:
> I think what we can do is to pass a list of channel objects into getAllLatestCommittedOffsetTokens?

Are you happy to unsafe cast from `SnowflakeStreamingIngestChannel` to `SnowflakeStreamingIngestChannelInternal`?